### PR TITLE
Move include/threads to src/

### PR DESCRIPTION
**Pull request by zeroadam**
Move threads functions into Item class

related to #3878

nothing found in addons

### CODE REVIEW

**Comment by MrPetovan** [comment]
I was thinking the functions you moved should be in a `Model\Thread` class of their own to make `Model\Item` lighter, but maybe @annando can convince me otherwise depending on the philosophy behind threads.

**Comment by zeroadam** [comment]
I was just going off the ethercalc suggestions. I believe @annando may have made that suggestion there.

**Comment by annando** [comment]
In - hopefully - the not so distant future the "thread" table will be as obsolete like the "item" table and had been replaced with other tables. So it is better - from my point of view - to have it all in a single class.

**Comment by MrPetovan** [comment]
Fair enough, let's proceed this way.

File: src/Model/Item.php [hunk 1]
```diff
@@ -1873,4 +1872,119 @@ public static function performLike($item_id, $verb)
...
+		$item['iid'] = $itemid;
+	
+		if (!$onlyshadow) {
+			$result = dba::insert('thread', $item);
+	
+			logger("Add thread for item ".$itemid." - ".print_r($result, true), LOGGER_DEBUG);
+		}
+	}
+
+	public static function updateThreadFromUri($itemuri, $uid)
```
Comment (MrPetovan) [inline src/Model/Item.php]: Please rename to `updateThreadByUri`

File: src/Model/Item.php [hunk 2]
```diff
@@ -1873,4 +1872,119 @@ public static function performLike($item_id, $verb)
...
+		$result = dba::update(
+			'item',
+			['title' => $item['title'], 'body' => $item['body'], 'rendered-html' => $item['rendered-html'], 'rendered-hash' => $item['rendered-hash']],
+			['id' => $items['id']]
+		);
+
+		logger("Updating public shadow for post ".$items["id"]." - guid ".$item["guid"]." Result: ".print_r($result, true), LOGGER_DEBUG);
+	}
+	
+	public static function deleteThreadFromUri($itemuri, $uid)
```
Comment (MrPetovan) [inline src/Model/Item.php]: Please rename to `deleteThreadByUri`

**Comment by MrPetovan** [comment]
Now you just need to resolve the conflicts from the earlier merge!

**Comment by zeroadam** [comment]
rebase and resolve conflicts

**Comment by AlfredSK** [comment]
I'm getting a lot of PHP fatal errors. I think they are related to this PR.

```
PHP Fatal error:  Uncaught Error: Cannot use object of type PDOStatement as array in /var/www/html/libranet/src/Model/Item.php:1981
Stack trace:
#0 /var/www/html/libranet/src/Model/Item.php(155): Friendica\Model\Item::deleteThread('4047729', 'fab_xxx@framxxx...')
#1 /var/www/html/libranet/src/Model/Item.php(1629): Friendica\Model\Item::delete('4047729', 40)
#2 /var/www/html/libranet/src/Worker/Expire.php(46): Friendica\Model\Item::expire('335', '90')
#3 [internal function]: Friendica\Worker\Expire::execute(335)
#4 /var/www/html/libranet/src/Core/Worker.php(349): call_user_func_array('Friendica\\Worke...', Array)
#5 /var/www/html/libranet/src/Core/Worker.php(243): Friendica\Core\Worker::execFunction(Array, 'Expire', Array, true)
#6 /var/www/html/libranet/src/Core/Worker.php(105): Friendica\Core\Worker::execute(Array)
#7 /var/www/html/libranet/scripts/worker.php(57): Friendica\Core\Worker::processQueue(false)
#8 {main}
  thrown in /var/www/html/libranet/src/Model/Item.php on line 1981
```

**Comment by MrPetovan** [comment]
I see the problem, I'll submit a patch shortly.

**Pull request merged**